frontend/dockerfile/builder/build.go

```diff
@@ -874,6 +874,13 @@ func contextByName(ctx context.Context, c client.Client, name string, platform *
 			st = &httpst
 		}
 		return st, nil, nil, nil
+	case "oci-layout":
+		ref := strings.TrimPrefix(vv[1], "//")
+		st := llb.OCILayout(ref,
```

Comment:
No, this is not ref. ContentStoreID and digest.

Reply:
I don't get that. What do you mean by contentstoreID?

From the end user perspective, all I do is pass "alias image foo to an oci layout at path x with hash abc123", using opt context. 

Internally, the client passes this to the frontend, which translates "foo" to a reference to an OCI layout path and hash (the root manifest for that image).

Where does this become an ID? How is it referenced, both in passing to remote components (frontend and buildkitd), and when buildkitd passes it back up the channel to the client?

Reply:
contentstoreID is the key in the `map[string]content.Store` that is exposed on the client side. Unique key that is used in the session request https://pkg.go.dev/github.com/moby/buildkit@v0.10.3/session/content#NewCallerStore. Client side session handler receives the request and by ContentStoreID finds the correct `content.Store` and by digest the correct root object from that `content.Store`.

Reply:
So when client side sees, eg, 3 opts with context, each pointing to a different source, it gives each one some unique ID, which is passed on to the builder. 

Is there an example of that being done? I figure that local must have it?

Reply:
https://github.com/moby/buildkit/blob/master/client/solve.go#L466

Reply:
Oh ok, now I see it. So the contentStoreID in the local case is just `local:/path/to/store`. This is what gets passed to the frontend to use in place of whatever image is aliases (e.g. `foo`), and what is sent to buildkitd, and in turn is then sent back to the client via the channel.

In the local case, a directory is enough, as it is just one thing, so `local:` (this is a local path) and `<dir>` (where to get it).
In the oci-layout case, we would need to indicate `oci-layout:` (this is an OCI layout) and `<dir>` where to get it and `<hash>` (what to extract from that layout. Roughly similar in concept to what we do with docker `docker-image:` followed by registry (where to get it) and ref (which to pull).

Reply:
Not sure I got it entirely. See above with the signature.